addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/handler/AccountHandler.java

```diff
@@ -0,0 +1,635 @@
+        }
+        refreshLogin = scheduler.scheduleWithFixedDelay(() -> {
+            checkLogin();
+        }, 0, 60, TimeUnit.SECONDS);
+
+        if (refreshJob != null) {
+            refreshJob.cancel(false);
+        }
+        refreshJob = scheduler.scheduleWithFixedDelay(() -> {
+            refreshData();
+        }, 4, config.pollingIntervalInSeconds, TimeUnit.SECONDS);
+
+        logger.debug("amazon account bridge handler started.");
+    }
+
+    private void checkLogin() {
+
+        synchronized (synchronizeConnection) {
+            Connection temp = connection;
+            if (temp == null) {
+                return;
+            }
+            Date loginTime = temp.tryGetLoginTime();
+            Date currentDate = new Date();
+            long currentTime = currentDate.getTime();
+            if (loginTime != null && currentTime - loginTime.getTime() > 3600000) // One hour
+            {
+                try {
+                    if (!temp.verifyLogin()) {
+                        temp.logout();
+                    }
+                } catch (IOException | URISyntaxException e) {
+                    logger.info("logout failed: {}", e.getMessage());
+                    temp.logout();
+                }
+            }
+            loginTime = temp.tryGetLoginTime();
+            if (loginTime != null && currentTime - loginTime.getTime() > 86400000 * 5) // 5 days
+            {
+                // Recreate session
+                this.stateStorage.storeState("sessionStorage", "");
+                temp = new Connection(temp.getEmail(), temp.getPassword(), temp.getAmazonSite(),
+                        this.getThing().getUID().getId());
+            }
+            boolean loginIsValid = true;
+            if (!temp.getIsLoggedIn()) {
+                try {
+
+                    // read session data from property
+                    String sessionStore = this.stateStorage.findState("sessionStorage");
+
+                    // try use the session data
+                    if (!temp.tryRestoreLogin(sessionStore)) {
+                        // session data not valid -> login
+                        int retry = 0;
+                        while (true) {
+                            try {
+                                temp.makeLogin();
+                                break;
+                            } catch (ConnectionException e) {
+                                // Up to 2 retries for login
+                                retry++;
+                                if (retry >= 2) {
+                                    temp.logout();
+                                    throw e;
+                                }
+                                // give amazon some time
+                                try {
+                                    Thread.sleep(2000);
+                                } catch (InterruptedException exception) {
+                                    // throw the original exception
```

Comment:
Is this really a delayed re-throw, why?

Reply:
No, the sleep is only for the while loop, which retries the operation, only if the sleep itself is interrupted, I want get the original thrown exception. So I think this is correct.